### PR TITLE
Dev Guide improvements.

### PR DESCRIPTION
**Pull request by pali6**
## About the PR
Clarified and added quite a lot of information to the dev guide. Co-authored with @Enfaeutchie .
I'd really appreciate if someone could doublecheck the information because I bet it's not all acurate.

## Why's this needed?
To streamline the process of contriburing to the game as much as possible.

### CODE REVIEW

**Comment by ZeWaka** [comment]
Added a bunch of comments to your WIP doc.

File: .github/DEV_GUIDE.md [hunk 1]
```diff
@@ -177,20 +275,79 @@ Wow! Great job on making your PR. :tada:
...
+
+**Merge Conflict:** When two branches made different changes to the same part of the code since they split off they can't be merged automatically. This is called a merge conflict and someone (likely you) needs to decide how to join these changes and resolve it manually.
 
 **Origin:** In this guide, this refers to your fork of the goonstation repo.
 
 **Upstream:** In this guide, this refers to the master Goonstation repo at https://github.com/goonstation/goonstation
 
+**Staged files:** Files marked to get commited.
+
 *[repo]: Repository - contains all Goonstation code and etc.
```
Comment (ZeWaka) [inline .github/DEV_GUIDE.md:286]: Move the annotations to the bottom?

File: .github/DEV_GUIDE.md [hunk 2]
```diff
@@ -6,6 +6,8 @@
 
 ### Step 1: Downloading Visual Studio Code :arrow_down: 
 
+Visual Studio Code is a free program you can use to edit and manage your code, as well as connect with GitHub to publish it. You can also use BYOND's built-in DreamMaker software to edit, though you'll probably still find VS Code useful.
+
```
Comment (ZeWaka) [inline .github/DEV_GUIDE.md:10]: ?

File: .github/DEV_GUIDE.md [hunk 3]
```diff
@@ -105,7 +122,32 @@ Here's the changes I'm making for the purpose of this guide:
 
 Now, save your changes. If we look at the Source Control tab, we'll see that we have some new changes. Git has found every change you made to your fork's repo on your computer! Even if you change a single space in a single line of code, Git will find that change. Just make sure you save your files.
 
-### Step 3: Up On Stage :movie_camera: 
+### Step 3: Testing your code :game_die:
+
+The easiest way to test your changes is to press F5. This compiles your code, runs the server and connects you to it, as well as automatically giving you admin permissions. It also starts a debugger that will let you examine what went wrong when a runtime error happens.
```
Comment (ZeWaka) [inline .github/DEV_GUIDE.md:127]: Mention CTRL+F5 (no debugger DS launch)?

Comment (pali6) [reply .github/DEV_GUIDE.md:127]: I had no idea about that one, thanks!

**Comment by pali6** [comment]
Also if you want Meld referenced there please add it yourself as I haven't really used it and literally all I know about it is that it can do three way merges.

**Comment by ZeWaka** [comment]
> Also if you want Meld referenced there please add it yourself as I haven't really used it and literally all I know about it is that it can do three way merges.

I'll probably add it at a later date whenever someone experiences a complicated merge.

**Pull request merged**